controllers/storagecluster/reconcile.go

```diff
@@ -434,6 +434,12 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			statusutil.SetErrorCondition(&instance.Status.Conditions, reason, message)
 
 			if instance.Status.Phase != statusutil.PhaseOnboarding {
+				// if the error was due to skipped storage classes
+				// instead of error phase we will set it to progressing
+				if strings.Contains(returnErr.Error(), "some StorageClasses were skipped while waiting for pre-requisites to be met") {
```

Comment:
Thought: Can we create a constant error string and use the same at both places? It will reduce the chances of someone changing the error and having the bug again. WDYT?

Reply:
Hmm, Yeah agreed. It can be a good step towards avoiding this bug again. Made the change. PTAL.